Dockerfile_ubuntu_deps

```diff
@@ -1,79 +1,95 @@
-    apt-get update
-RUN apt-get install -y --no-install-recommends \
-        build-essential \
-        git \
-        wget \
-        unzip \
-        yasm \
-        pkg-config \
-        libtool \
-        libssl-dev \
-        nasm \
-        automake \
-        gfortran
-
-# rm -rf /var/lib/apt/lists/*
+RUN . ./etc/os-release && \
+	apt-get update && \
+	DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+		wget \
+		software-properties-common && \
+	wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc | gpg --dearmor - > /etc/apt/trusted.gpg.d/kitware.gpg && \
+	apt-add-repository "deb https://apt.kitware.com/ubuntu/ $UBUNTU_CODENAME main" && \
+	apt-get clean && \
+	apt-get update && \
+	DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+		build-essential \
+		git \
+		unzip \
+		yasm \
+		pkg-config \
+		libgmp-dev \
+		libmpfr-dev \
+		libtool \
+		libssl-dev \
+		nasm \
+		automake \
+		cmake \
+		gfortran
 
 ENV AV_DEV=/opt/AliceVision_git \
     AV_BUILD=/tmp/AliceVision_build \
     AV_INSTALL=/opt/AliceVision_install \
-    AV_BUNDLE=/opt/AliceVision_bundle \
     PATH="${PATH}:${AV_BUNDLE}"
 
-# Manually install cmake
-WORKDIR /tmp/cmake
-ENV CMAKE_VERSION=3.17
-ENV CMAKE_VERSION_FULL=${CMAKE_VERSION}.2
-RUN wget https://cmake.org/files/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION_FULL}.tar.gz && \
-    tar zxf cmake-${CMAKE_VERSION_FULL}.tar.gz && \
-    cd cmake-${CMAKE_VERSION_FULL} && \
-    ./bootstrap --prefix=/usr/local  -- -DCMAKE_BUILD_TYPE:STRING=Release -DCMAKE_USE_OPENSSL:BOOL=ON && \
-    make -j$(nproc) install && \
-    cd /tmp && \
-    rm -rf cmake
-
-COPY CMakeLists.txt "${AV_DEV}/"
+COPY dl/vlfeat_K80L3.SIFT.tree ${AV_INSTALL}/share/aliceVision/
+RUN echo "export ALICEVISION_VOCTREE=${AV_INSTALL}/share/aliceVision/vlfeat_K80L3.SIFT.tree" > /etc/profile.d/alicevision.sh
 
+COPY CMakeLists.txt ${AV_DEV}/
 
-# Cannot get rig of this error on lapack build on Ubuntu, so use system libraries for lapack/suitesparse:
-#CMake Error at BLAS/SRC/cmake_install.cmake:53 (file):
-#  file INSTALL cannot find
-#  "/tmp/AliceVision_build/external/lapack_build/lib/libblas.so.3.8.0".
-#Call Stack (most recent call first):
-#  BLAS/cmake_install.cmake:46 (include)
-#  cmake_install.cmake:72 (include)
-RUN apt-get  install -y --no-install-recommends liblapack-dev libsuitesparse-dev
+COPY dl/deps ${AV_BUILD}/external/download/
 
 WORKDIR "${AV_BUILD}"
 RUN cmake "${AV_DEV}" \
      -DCMAKE_BUILD_TYPE=Release \
      -DALICEVISION_BUILD_DEPENDENCIES:BOOL=ON \
      -DAV_BUILD_ALICEVISION:BOOL=OFF \
-     -DAV_BUILD_LAPACK:BOOL=OFF \
-     -DAV_BUILD_SUITESPARSE:BOOL=OFF \
      -DCMAKE_INSTALL_PREFIX="${AV_INSTALL}"
 
+# Symlink lib64 to lib as qtOIIO expects to find OIIO in lib64
+RUN mkdir -p "${AV_INSTALL}/lib" && \
+        ln -s lib "${AV_INSTALL}/lib64"
+
 WORKDIR "${AV_BUILD}"
-RUN make && cd /opt && rm -rf ${AV_BUILD}
+RUN make -j"$(nproc)" zlib
```

Comment:
It could be useful for debugging, so we can keep it in comment. But if we launch the global `make -j$(nproc)` without specific target, it will be able to build multiple projects in parallel.
And it would also avoid a RUN storing the status at each command.

Reply:
I observed some issues with make not passing the job count down to child makes in this circumstance. This may have been before the other changes though, so I'll retest and amend if the job count is inherited correctly.

Reply:
I've confirmed that the jobs are passed down correctly now, so I'll update this to use a single global make

Reply:
FYI, adding --rm to the `docker build` command instructs Docker to remove intermediate images if the build completes successfully.

This change is still good though, as it increases the potential for parallelism in the build.

Reply:
Yes, but I use these docker images to upload the docker images of the releases on dockerhub, and also the WIP image of dependencies for CI. So the size of the docker image is still important.